nshlib/nsh_syscmds.c

```diff
@@ -579,6 +581,8 @@ int cmd_uname(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
   /* Process each option */
 
   first = true;
+  lib_memoutstream(&stream, alloca(sizeof(struct utsname)),
```

Comment:
is it enough to allocate struct utsname in the extreme case? since we add more ' ' and '\n'.

Reply:
It is adequate, an extra 7 characters are added when using “uname -a”
But utsname->sysname is 21 (SYS_NAMELEN) bytes, only 6 bytes ("NuttX") are actually used